contracts/core/Registry.sol

```diff
@@ -21,27 +36,75 @@ contract Registry {
     // attestationAddr = addr(hash(id, name, owner))
     // OR deploy right away with CREATE2 as proxy so they can do anything (or just forward ERC20s and ETH)
     // id => IdentityDetails
-    function createIdentity(IdentityDetails memory _identityDetails, address[] memory _owners) external returns (uint256) {
 
+    // ASK: changing the function signature as attestationAddr has to be generated from the name and caller
+    /// @notice Creates a new identity
+    /// @dev This will also set the attestation address generated from msg.sender and name
+    function createIdentity(
+        string memory name,
+        Metadata memory metadata,
+        address[] memory _owners
+    ) external returns (uint256) {
+
+        IdentityDetails memory identityDetails = IdentityDetails(
+            name,
+            metadata,
+            // ASK : should identityId be used as the hash for attestationAddr?
+            // cause project can't have same attestAddr across chains
+            generateAttestationAddr(name, msg.sender)
+        );
+
+        identities[identityId++] = identityDetails;
+
+        for(uint i = 0; i < _owners.length; i++) {
+            _grantRoles(_owners[i], identityId);
+        }
+        
+        emit IdentityCreated(identityId, name, metadata, identityDetails.attestationAddr);
+        
+        return identityId;
+    }
+
+    /// @notice Updates the name of the identity
+    /// @dev This will also change the attestation address, since it's a hash that includes the name
+    function updateIdentityName(uint _identityId, string memory _name) onlyRoles(identityId) external {
+        identities[_identityId].name = _name;
+        identities[_identityId].attestationAddr = generateAttestationAddr(_name, msg.sender);
+
+        emit IdentityNameUpdated(_identityId, _name);
     }
 
-    // updates the name of the identity
-    // note that this will also change the attestation address, since it's a hash that includes the name
-    function updateIdentityName(uint _identityId, string memory _name) external {
+    function changeCoreOwner(uint _identityId, address _newCoreOwner) external {
```

Comment:
Not sure how to track core owner of a identity separately. 
If we store it IdentityDetails, then this function is needed 

@zobront

Reply:
Yup agreed. Dropping this for now